lib/src/utils/accent_color_listener.dart

```diff
@@ -0,0 +1,160 @@
+///
+/// ```dart
+/// StreamBuilder(
+///   stream: AccentColorListener.instance.onChangedStream,
+///     builder: (context, _) {
+///       final AccentColor? accentColor =
+///           AccentColorListener.instance.currentAccentColor;
+///
+///       return SomeWidget(
+///         accentColor: accentColor,
+///         child: …
+///       );
+///     },
+///  );
+/// ```
+class AccentColorListener {
+  /// A shared instance of [AccentColorListener].
+  static final instance = AccentColorListener();
+
+  /// A map which maps hue components of the [UiElementColor.controlAccentColor]
+  /// color captured with the [NSAppearanceName.aqua] appearance in the
+  /// [NSColorSpace.genericRGB] color space to the corresponding [AccentColor].
+  static final hueComponentToAccentColor = <double, AccentColor>{
+    0.6085324903200698: AccentColor.blue,
+    0.8285987697113538: AccentColor.purple,
+    0.9209523937489168: AccentColor.pink,
+    0.9861913496946438: AccentColor.red,
+    0.06543037411201169: AccentColor.orange,
+    0.11813830353929083: AccentColor.yellow,
+    0.29428158007138466: AccentColor.green,
+    0.0: AccentColor.graphite,
+  };
+
+  /// The currently active accent color.
+  AccentColor? _currentAccentColor;
+
+  /// The currently active accent color.
+  AccentColor? get currentAccentColor => _currentAccentColor;
+
+  /// Notifies listeners when the accent color changes.
+  final _accentColorStreamController = StreamController<void>.broadcast();
+
+  /// Streams the user’s system accent color selection.
+  ///
+  /// Emits a new value whenever the system accent color selection changes.
+  Stream<void> get onChangedStream => _accentColorStreamController.stream;
+
+  /// A stream subscription for the [SystemColorObserver] stream.
+  StreamSubscription<void>? _systemColorObserverStreamSubscription;
+
+  /// Initializes this class.
+  void _init() {
+    if (kIsWeb) return;
+    if (!Platform.isMacOS) return;
+
+    _initCurrentAccentColor();
+    _initSystemColorObserver();
+  }
+
+  /// Disposes this listener.
+  void dispose() {
+    _systemColorObserverStreamSubscription?.cancel();
+  }
+
+  /// Initializes the current accent color. This method is to be called whenever
+  /// a change is detected.
+  Future<void> _initCurrentAccentColor() async {
+    final hueComponent = await _getHueComponent();
+    _currentAccentColor = _resolveAccentColorFromHueComponent(hueComponent);
+    _accentColorStreamController.add(null);
+  }
+
+  /// Initializes the current system color observer. This method may only be
+  /// called once.
+  void _initSystemColorObserver() {
+    assert(_systemColorObserverStreamSubscription == null);
+
+    _systemColorObserverStreamSubscription =
+        AppkitUiElementColors.systemColorObserver.stream.listen((_) {
+      _initCurrentAccentColor();
+      _accentColorStreamController.add(null);
+    });
+  }
+
+  /// Returns the hue component of the currently active accent color on macOS.
```

Comment:
```suggestion
  /// Returns the hue component of the active accent color selection on macOS.
```
"Active" already implies currency